module/VuFind/src/VuFind/AjaxHandler/DoiLookup.php

```diff
@@ -118,4 +155,41 @@ public function handleRequest(Params $params)
         }
         return $this->formatResponse($response);
     }
+
+    /**
+     * Proxify any DOI icon links
+     *
+     * @param array $dois DOIs
+     *
+     * @return array
+     */
+    protected function proxifyIconLinks(array $dois): array
+    {
+        if (!$this->proxyIcons) {
+            return $dois;
+        }
+
+        $serverHelper = $this->viewRenderer->plugin('serverurl');
```

Comment:
I realize that it _should_ be impossible to reach this code without using the updated factory, but maybe it's worth throwing an exception of the viewRenderer is missing, just to be on the safe side.

Reply:
Fixed and then I realized that local icons were not properly rendered when proxying was disabled. Had to revise so that the method does everything when view renderer is available and nothing otherwise. That's not quite optimal, so I'm wondering if a BC break would be better in this case.

Reply:
Yeah, maybe a BC break is justified in this case. I'm rather doubtful that many (any?) people have customized this, so it's unlikely to have much impact... and then we can have less confusing code.

Reply:
I've changed the code to always expect new constructor style. I also added test coverage for the new functionality.